main.go

```diff
@@ -5,61 +5,110 @@ import (
 	"context"
 	"errors"
 	"fmt"
-	"io/ioutil"
+	"io"
+	"log/slog"
 	"net/http"
 	"os"
+	"slices"
 	"strings"
 
-	"github.com/caarlos0/env/v6"
-	egoScaleLog "github.com/exoscale-labs/snap-o-matic/log"
-	"github.com/exoscale/egoscale"
 	flag "github.com/spf13/pflag"
-	log "gopkg.in/inconshreveable/log15.v2"
+
+	v3 "github.com/exoscale/egoscale/v3"
+	"github.com/exoscale/egoscale/v3/credentials"
 )
 
 const (
 	metadataURL = "http://169.254.169.254/latest/meta-data/vm-id"
-	autosnapTag = "autosnap"
 
 	defaultSnapshotsRetention = 7
-)
-
-var (
-	exo *egoscale.Client
-	cfg config
+	defaultEndpoint           = v3.CHGva2
```

Comment:
By default, v3 returns a client in gva2 (according to API spec), but it sounds you are using it for the usage.

Reply:
good to know, but yes in this case it wouldn't make the code simpler if I relied on the v3 default.